packages/core/src/components/ui/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button as UIButton, InputField as UIInputField } from '@faststore/ui'
+import type { ComponentPropsWithRef, FormEvent, ReactNode } from 'react'
+import { forwardRef, useRef } from 'react'
+
+import { Icon, useUI } from '@faststore/ui'
+import Link from 'src/components/ui/Link'
+import { useNewsletter } from 'src/sdk/newsletter/useNewsletter'
+
+export interface NewsletterProps
+  extends Omit<ComponentPropsWithRef<'form'>, 'title' | 'onSubmit'> {
+  /**
+   * Icon for the section.
+   */
+  icon: {
+    icon: string
+    alt: string
+  }
+  /**
+   * Title for the section.
+   */
+  title: ReactNode
+  /**
+   * A description for the section.
+   */
+  description?: ReactNode
```

Comment:
I think these props are strings now that comes from CMS.
```suggestion
  title: string
  /**
   * A description for the section.
   */
  description?: string
```